qa/suites/fs/upgrade/featureful_client/old_client/tasks/2-upgrade.yaml

```
@@ -21,7 +21,9 @@ tasks:
 - print: "**** done mds pre-upgrade sequence"
 - install.upgrade:
     mon.a:
+      branch: quincy
```

```suggestion
      branch: quincy
          mon.a:
          mon.b:      
```
Can something like this work?

---

probably, idk. my yaml knowledge is pretty limited and I followed the following

```
qa/suites/teuthology/multi-cluster/all/upgrade.yaml:34:- install.upgrade:
qa/suites/teuthology/multi-cluster/all/upgrade.yaml-35-    ceph.mon.a:
qa/suites/teuthology/multi-cluster/all/upgrade.yaml-36-      branch: jewel
qa/suites/teuthology/multi-cluster/all/upgrade.yaml-37-    backup.mon.a:
qa/suites/teuthology/multi-cluster/all/upgrade.yaml-38-      branch: jewel
```

---

venky's suggestion is the correct approach to specify upgrade to a specific branch

---

yeah this works, i was just thinking about reducing the number of times `branch: quincy` happens in the yaml.

---

From the teuthology source code and comments there only have two config formats:

When upgrading all `roles` we can just do:
```
        tasks:
        - install.upgrade:
             all:
                branch: end        
```

    or must specify specific `roles` like:
```
        tasks:
        - install.upgrade:
             mon.a:
                branch: end
             osd.0:
                branch: other
```

---

So here @vshankar is correct!